UI Layer (Alphafold Portal)

yudyhendry opened the pull request:
- Add frontend and backend components on top of existing AF repo
        - Managing pipeline job runs from UI Dashboard
        - Create new pipeline AF Pipeline Job from UI
        - Add user label as creator of the pipeline job

Commit:
UI Layer (Alphafold Portal)

        - Add frontend and backend components on top of existing AF repo
        - Managing pipeline job runs from UI Dashboard
        - Create new pipeline AF Pipeline Job from UI
        - Add user label as creator of the pipeline job

Commit:
further readme update and tf test

Commit:
remove Step 4 of building kfp images (cuda111)

- the steps are part of terraform deployment

Commit:
add required compile-time files

Commit:
elaborate READMEs

Commit:
expose RELAX machineType, count, and auto set accel.type

Commit:
Fix error not passing accelerator count correctly

Commit:
auto-calculate accelerator count

- remove entry fields for accelerator count

Commit:
expose all candidates PDB file structures in the portal

Commit:
add more hint about the GPU Type

Commit:
auto close dialog and refresh and scrollable background when drawers out

Commit:
Fix deployment error (tf and react build)

Commit:
fix react build

Commit:
Expose Top Prediction and Top Relaxation bucket location in Portal

Commit:
Fix dashboard showing post login info

Commit:
fix dashboard view when pipelines job empty

Commit:
fix new project tf apply error

Commit:
Add features:

- auto populate protein type
- show protein type and residue
- set default values for smallBFD, predictionCount, relaxation
- hide default values under advanced setting group
- introduce Run Tag to group Experiment IDs

Commit:
Add new features on table:

- table able to group column(s)
- default grouping to Run Tag column
- enable sorting and filtering on the table
- better table loading effect

Commit:
fix transpiling error (failing tf)

Commit:
Fix fasta-check error

Commit:
Support legacy Container Registry by using is_gcr_io_repo flag

Commit:
Fix residue counting and auto populate machine spec

Commit:
update README.md for both AF1 and AF2 on IS_GCR_IO_REPO value usage

Commit:
Elaborate IS_GCR_IO_REPO explanation

Commit:
Resize columns and display Accuracy and their links

- prediction url link to GCS bucket
- relaxation url link to GCS bucket
- show in descending ranking_confidence (Accuracy)

Commit:
fix CI checks error

Commit:
fix relax uri extraction

chmstimoteo commented on .dockerignore:
Why these lines are excluded?

chmstimoteo commented on .gitignore:
Why these lines are excluded?

chmstimoteo commented on env-setup-portal/Dockerfile:
Pls move this `Dockerfile` file to the root folder and rename it as Dockerfile.frontend or something. Let's keep all docker files in a single location.

chmstimoteo commented on env-setup-portal/alphafold-cloudbuild.yml:
pls move the cloud build recipe file to the same location where the docker files should be.

chmstimoteo commented on env-setup-portal/backend/T1031.fasta:
This sample sequence is already provided in the `sequences/` folder.

chmstimoteo commented on env-setup-portal/backend/main.py:
Pls, move the whole `backend/` folder inside the `src/` folder. So that the folder hierarchy becomes `src/backend/`.

chmstimoteo commented on env-setup-portal/frontend/.eslintrc.cjs:
Pls, move the whole `frontend/` folder inside the `src/` folder. So that the folder hierarchy becomes `src/frontend/`.

chmstimoteo commented on env-setup-portal/src/__init__.py:
The folder `src/`, should not have been moved to `env-setup-portal/src/`.

chmstimoteo replied:
I did a quick comparison in between the files in `src/` and `env-setup-portal/src/`. I cannot find any breaking changes. So it is ideal you reference the code in `src/` and delete the the whole folder `env-setup-portal/src/`.

[src_changes_diff.txt](https://github.com/GoogleCloudPlatform/vertex-ai-alphafold-inference-pipeline/files/13835299/src_changes_diff.txt)

chmstimoteo reviewed (changes requested):
initial changes asked.

chmstimoteo commented on env-setup-portal/README.md:
Also, as a best practice, we will work with the concept of terraform modules.
So that we don't need to copy terraform.tfvars files.

chmstimoteo replied:
We still need this.. look how we did for [MAJ](https://github.com/GoogleCloudPlatform/marketing-analytics-jumpstart/tree/main/infrastructure/terraform)

Commit:
dynamically support gcr.io (legacy repo)

Commit:
fix Cloud Run image name

Commit:
Bulk refactor frontend, backend, and Dockerfiles

Commit:
bulk fix module imports error after refactor

Commit:
Adapt AF1 to use yml config file build

chmstimoteo commented on Alphafold2.dockerfile:
Please call this file something like `Alphafold_portal.dockerfile`.

chmstimoteo commented on alphafold-portal.yml:
```diff
@@ -0,0 +1,11 @@
+  args: [ 'build',
+          '.',
+          '-f',
+          './Alphafold2.dockerfile',
```
rename it..

chmstimoteo reviewed (approved):
Last couple changes before merge.

Commit:
Feature: protein visualization

Commit:
add citation to 3Dmol

Commit:
renamed dockerfiles,ymls and update README for tfvars

Commit:
Add clarity on AF Portal discovery in READMEAdd clarity on AF Portal
discovery in READMEAdd clarity on AF Portal discovery in READMEAdd
clarity on AF Portal discovery in READMEAdd clarity on AF Portal
discovery in READMEAdd clarity on AF Portal discovery in READMEAdd
clarity on AF Portal discovery in READMEAdd clarity on AF Portal
discovery in READMEAdd clarity on AF Portal discovery in README

Commit:
cross tool run fix license header
